openmetadata-service/src/main/java/org/openmetadata/service/apps/AbstractNativeApplication.java

```diff
@@ -0,0 +1,83 @@
+package org.openmetadata.service.apps;
+
+import static org.openmetadata.service.apps.scheduler.AppScheduler.APP_INFO;
+import static org.openmetadata.service.apps.scheduler.AppScheduler.COLLECTION_DAO_KEY;
+
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.AppRuntime;
+import org.openmetadata.schema.entity.app.AppType;
+import org.openmetadata.schema.entity.app.Application;
+import org.openmetadata.schema.entity.app.ScheduleType;
+import org.openmetadata.schema.entity.app.ScheduledExecutionContext;
+import org.openmetadata.service.apps.scheduler.AppScheduler;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.util.JsonUtils;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+@Slf4j
+public class AbstractNativeApplication implements NativeApplication {
+  private CollectionDAO collectionDAO;
+  private Application app;
+
+  @Override
+  public void init(Application app, CollectionDAO dao) {
+    this.collectionDAO = dao;
+    this.app = app;
+  }
+
+  @Override
+  public void triggerOnDemand(Object requestObj) {
+    // Validate Native Application
+    if (app.getScheduleType().equals(ScheduleType.Scheduled)) {
+      AppRuntime runtime = getAppRuntime(app);
+      validateServerExecutableApp(runtime);
+      // Trigger the application
+      AppScheduler.getInstance().triggerOnDemandApplication(app);
+    } else {
+      throw new IllegalArgumentException("Live Application cannot scheduled.");
```

Comment:
wrap it around CatalogApplicationMessage

Reply:
done